src/client/datascience/raw-kernel/rawKernel.ts

```
@@ -104,9 +104,10 @@ export class RawKernel implements Kernel.IKernel {
         socket.emit('open');
     }
 
-    public shutdown(): Promise<void> {
+    public async shutdown(): Promise<void> {
         suppressShutdownErrors(this.realKernel);
-        return this.realKernel.shutdown().then(() => this.kernelProcess.dispose());
+        await this.kernelProcess.dispose();
+        this.socket.dispose();
```

@rchiodo @IanMatthewHuff Changed here as previous implementation would fall over.
We cannot call `shutdown` of the `realKernel` cuz it tries to make http call to jupyter.

---

Yeah but it just fails silently. I thought it also set state on itself. This would mean we're not setting that state.

---
In reply to: [417673182](https://github.com/microsoft/vscode-python/pull/11487#discussion_r417673182) [](ancestors = 417673182)